### DISCUSSION TITLE
RPC GPU Offload Tracing

### DISCUSSION THREAD

**Discussion by Allan-Luu** [ggml-org/llama.cpp#9518] [Q&A] [answered]
Hello,

I currently have GPUs on a cluster that are receiving offloaded layers from my llama.cpp on a different host through RPC servers. I'm trying to see which process is offloading the LLM inference onto the GPUs and trace the GPU events triggered by llama.cpp.

Any advice on how to trace the GPU events triggered by RPC servers?

**Comment by rgerganov** [comment, marked as answer]
You can trace the calls to `rpc-server` by setting `GGML_DEBUG` to `1` [here](https://github.com/ggerganov/llama.cpp/blob/master/ggml/src/ggml-rpc.cpp#L32).

**Reply by Allan-Luu** [reply]
Is there an option for this to trace calls from local GPUs from host machine as well?
